React/Views/RCTScrollView.m

```diff
@@ -602,6 +602,32 @@ - (void)scrollToOffset:(CGPoint)offset animated:(BOOL)animated
   }
 }
 
+/**
+ * If this is a vertical scroll view, scrolls to the bottom.
+ * If this is a horizontal scroll view, scrolls to the right.
+ */
+- (void)scrollToEnd:(BOOL)animated
+{
+  BOOL isHorizontal = (_scrollView.contentSize.width > self.frame.size.width);
```

Comment:
It seems we're repeating this logic. Can you move it to a private helper method?